src/amo/constants.js

```diff
@@ -1,13 +1,37 @@
+/* @flow */
 // Action types.
-export const SHOW_EDIT_REVIEW_FORM = 'SHOW_EDIT_REVIEW_FORM';
-export const SHOW_REPLY_TO_REVIEW_FORM = 'SHOW_REPLY_TO_REVIEW_FORM';
-export const FETCH_REVIEWS = 'FETCH_REVIEWS';
-export const HIDE_EDIT_REVIEW_FORM = 'HIDE_EDIT_REVIEW_FORM';
-export const HIDE_REPLY_TO_REVIEW_FORM = 'HIDE_REPLY_TO_REVIEW_FORM';
-export const SET_ADDON_REVIEWS = 'SET_ADDON_REVIEWS';
-export const SET_REVIEW = 'SET_REVIEW';
-export const SET_REVIEW_REPLY = 'SET_REVIEW_REPLY';
-export const SEND_REPLY_TO_REVIEW = 'SEND_REPLY_TO_REVIEW';
+export const SHOW_EDIT_REVIEW_FORM: 'SHOW_EDIT_REVIEW_FORM' =
```

Comment:
I hate to complain and I've been quiet about a lot of flow creeping into parts of the codebase that aren't proptypes, but this is baffling to look at. I get full flow coverage would mean never needing constants, but as we use them, forcing the flow coverage to the constant values is really unfriendly.

Reply:
It's not great but utilizing real action types adds a lot of value. It gives us safety checks in how each reducer case handles its action object. Reducer code is one of the most critical parts of the application. Our unit tests cannot always cover using invalid payload properties--those bugs are very easy to introduce when refactoring action code.

> a lot of flow creeping into parts of the codebase

This code was already Flow-typed, it just wasn't working.

Reply:
Fair enough. I think I'm just cranky about the syntax. I've had more coffee and now I'm fine with it, ignore me 😉

Reply:
Also, since our test code doesn't have Flow coverage, it would be risky to switch away from constants. If one of these constants got renamed, we might miss a test. Hopefully it would fail but sometimes tests do a lot of stubbing.